pkg/storer/reserve.go

```diff
@@ -621,25 +651,27 @@ func transformedAddressSOC(hasher *bmt.Hasher, chunk swarm.Chunk) (swarm.Address
 }
 
 type sampleStat struct {
-	TotalIterated      atomic.Int64
-	NotFound           atomic.Int64
-	NewIgnored         atomic.Int64
-	IterationDuration  atomic.Int64
-	GetDuration        atomic.Int64
-	HmacrDuration      atomic.Int64
-	ValidStampDuration atomic.Int64
+	TotalIterated       atomic.Int64
+	NotFound            atomic.Int64
+	NewIgnored          atomic.Int64
+	BelowBalanceIgnored atomic.Int64
```

Comment:
Pardon my ignorance, but what's the difference between "BelowBalanceIgnored" and "ValidStampDuration"?  I thought the stamp duration (TTL) was based on the balance of the stamp vs the current storage price?

Reply:
`BelowBalanceIgnored` is count of chunks that are ignored form sample calculation logic because it's batch balance value is not enough to cover storage cost until next round of shelling game.
`ValidStampDuration` is total time spent for stamp validation (in sample calculation logic). actually it should be this, but on new storage implementation this piece of logic is missing. [here](https://github.com/ethersphere/bee/blob/exclude-batches-close-to-expiry/pkg/_localstore/sampler.go#L234) is link to old sampling logic which has this metrics.

Reply:
Or am I completely mistaken that there ever was a minimum stamp lifetime in the determination of what should be hashed for the reserve calculations?   Maybe it was that new chunks were excluded because they may not be fully propagated into the swarm?

Reply:
> Or am I completely mistaken that there ever was a minimum stamp remaining lifetime in the determination of what should be hashed for the reserve calculations? Maybe it was that new chunks were excluded because they may not be fully propagated into the swarm?

Yes so till now we only excluded chunks based on timestamp. This was not fully correct but this was to mainly protect against new live syncing chunks which may not be propagated fully during the sampling. With this change, we will tighten the screws even more. The goal is to not be affected by stamp expirations during the sampling. So we will ignore batches that may not last on all nodes while they are sampling.